src/shared/services/bolt/cypherTypesFormatting.js

```diff
@@ -37,7 +37,7 @@ export const stringModifier = anything => {
 
 const numberFormat = anything => {
   // Exclude false positives and return early
-  if ([Infinity, -Infinity].includes(anything)) {
+  if ([Infinity, -Infinity, NaN].includes(anything)) {
     return `${anything}`
   }
   if (Math.floor(anything) === anything) {
```

Comment:
this wont handle `NaN`. The main reason being any strict equality comparisons with `NaN` will always return `false` according to the JS Spec. Another option could be to add in a `isNan` here too but given that we are using the array above to exclude false positives (typeof number), then it seems like a logical place...